### ISSUE TITLE
Latest version breaks interface mocks when property overridden in derived interface

### ISSUE THREAD

**Issue by abe545** [moq/moq4#162]
This was working in 4.2.1409.1722 , but is broken in 4.2.1502.911.

It is a fairly complex setup, but the idea is that you have a derived interface that wants to add a set method to an existing read-only property on the base interface:

``` cs
public interface A 
{
    decimal? Value { get; }
}

public interface B : A
{
    new decimal? Value { get; set; }
}
```

You used to be able to create a mock like this:

``` cs
var i = Mock.Of<B>(b => b.Value == .14M);
```

And `i.Value` would be `0.14`.
But now `i.Value` will be `null`.

**Comment by stakx** [comment]
This change has been introduced by commit 97a0f149a3369da8e3fd79895abd61e242e7fc84. Looking at it now, but I'm not sure yet what can or should be done here.

**Comment by stakx** [comment]
Fixed in Moq 4.7.58.
